src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -513,9 +513,7 @@ private async Task<TextDocument> UpdateDocumentInfoAsync(TextDocument document,
 
                 foreach (var kv in documentChecksums)
                 {
-                    Debug.Assert(assetProvider.EnsureCacheEntryIfExists(kv.Item2.Info), "Expected the prior call to GetAssetsAsync to obtain all items for this loop.");
-
-                    var info = await assetProvider.GetAssetAsync<DocumentInfo.DocumentAttributes>(kv.Item2.Info, cancellationToken).ConfigureAwait(false);
+                    var info = assetProvider.GetRequiredAsset<DocumentInfo.DocumentAttributes>(kv.Item2.Info);
```

Comment:
this is safe, and correct (which the debug assert was alreay checking).  on line 512 we requests all these doc-attribute assets for these checksums.  So the requirement is that they *must* now be here when we go fetch them.